### ISSUE TITLE
Fieldtype's `parent` not set when in Asset

### ISSUE THREAD

**Issue by edalzell** [statamic/cms#4291]
## Bug Description

A custom fieldtype, in an asset blueprint, has no `parent` field set, when using the Asset stack (like editing an asset in the CP).

## How to Reproduce

1. Create Fieldtype
2. in the `preload` method:

```php
        if (! $asset = $this->field->parent()) {
            return [];
        }

        return [
            'asset_url' => $asset->absoluteUrl(),
        ];
```
3. see that `$this->field->parent()` is null.

## Environment

```
Statamic 3.2.5 Pro
Laravel 8.61.0
PHP 8.0.10
rias/statamic-link-it 2.2.3
silentz/asset-url dev-main
```

**Comment by edalzell** [comment]
cc @jsblair9

**Comment by duncanmcclean** [comment]
Confirmed - it's `null`, whereas it should be an `Asset` instance.

**Comment by duncanmcclean** [comment]
Fixed by #8644.
